src/templates/summary/index.ts

```diff
@@ -1,19 +1,75 @@
-import { getProposals } from '../../helpers/snapshot';
+import removeMd from 'remove-markdown';
+import { getFollows, getProposals, getVotes, Space, Proposal } from '../../helpers/snapshot';
 import { formatShortDate } from '../../helpers/date';
 import buildMessage from '../builder';
 import constants from '../../helpers/constants.json';
 import type { TemplatePrepareParams } from '../../../types';
 
+type ProposalStatus = 'pending' | 'active' | 'closed';
+type GroupedSpaces = { space: Space; proposals: Proposal[] };
+type GroupedProposals = Record<ProposalStatus, GroupedSpaces[]>;
+
+const MAX_SHORT_BODY_LENGTH = 150;
+
+export async function getGroupedProposals(addresses: string[], startDate: Date, endDate: Date) {
+  const follows = await getFollows(addresses);
+  const proposals = await getProposals(
+    follows.map(follow => follow.space.id),
+    startDate,
+    endDate
+  );
+  const votes = await getVotes(
+    proposals.map(proposal => proposal.id),
+    addresses
+  );
+
+  const groupedProposals: GroupedProposals = {
```

Comment:
can we use `groupBy` for this task? https://lodash.com/docs/4.17.15#groupBy

Reply:
Can you propose something ? From my tests, using groupBy is longer than not using it, we're grouping by 2 different properties, and the final results have a different format from the one returned by groupBy